plugins/processors/reverse_dns/reversedns.go

```diff
@@ -0,0 +1,155 @@
+  ## It's probably best to keep this number fairly low.
+  max_parallel_lookups = 10
+
+  ## ordered controls whether or not the metrics need to stay in the same order
+  ## this plugin received them in. If false, this plugin will change the order
+  ## with requests hitting cached results moving through immediately and not
+  ## waiting on slower lookups. This may cause issues for you if you are
+  ## depending on the order of metrics staying the same. If so, set this to true.
+  ## keeping the metrics ordered may be slightly slower.
+  ordered = false
+
+  [[processors.reverse_dns.lookup]]
+    ## get the ip from the field "source_ip", and put the result in the field "source_name"
+    field = "source_ip"
+    dest = "source_name"
+
+  [[processors.reverse_dns.lookup]]
+    ## get the ip from the tag "destination_ip", and put the result in the tag
+    ## "destination_name".
+    tag = "destination_ip"
+    dest = "destination_name"
+
+    ## If you would prefer destination_name to be a field instead, you can use a
+    ## processors.converter after this one, specifying the order attribute.
+`
+
+type lookupEntry struct {
+	Tag   string `toml:"tag"`
+	Field string `toml:"field"`
+	Dest  string `toml:"dest"`
+}
+
+type ReverseDNS struct {
+	reverseDNSCache *ReverseDNSCache
+	acc             telegraf.Accumulator
+	parallel        parallel.Parallel
+
+	Lookups            []lookupEntry   `toml:"lookup"`
+	CacheTTL           config.Duration `toml:"cache_ttl"`
+	LookupTimeout      config.Duration `toml:"lookup_timeout"`
+	MaxParallelLookups int             `toml:"max_parallel_lookups"`
+	Ordered            bool            `toml:"ordered"`
+	Log                telegraf.Logger `toml:"-"`
+}
+
+func (r *ReverseDNS) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *ReverseDNS) Description() string {
+	return "ReverseDNS does a reverse lookup on IP addresses to retrieve the DNS name"
+}
+
+func (r *ReverseDNS) Start(acc telegraf.Accumulator) error {
+	r.acc = acc
+	r.reverseDNSCache = NewReverseDNSCache(
+		time.Duration(r.CacheTTL),
+		time.Duration(r.LookupTimeout),
+		r.MaxParallelLookups, // max parallel reverse-dns lookups
+	)
+	if r.Ordered {
+		r.parallel = parallel.NewOrdered(acc, r.asyncAdd, 10000, r.MaxParallelLookups)
+	} else {
+		r.parallel = parallel.NewUnordered(acc, r.asyncAdd, r.MaxParallelLookups)
+	}
+	return nil
+}
+
+func (r *ReverseDNS) Stop() error {
+	r.parallel.Stop()
+	r.reverseDNSCache.Stop()
+	return nil
+}
+
+func (r *ReverseDNS) Add(metric telegraf.Metric, acc telegraf.Accumulator) {
+	r.parallel.Enqueue(metric)
+}
+
+func (r *ReverseDNS) asyncAdd(metric telegraf.Metric) []telegraf.Metric {
+	for _, lookup := range r.Lookups {
+		if len(lookup.Field) > 0 {
+			if ipField, ok := metric.GetField(lookup.Field); ok {
+				if ip, ok := ipField.(string); ok {
+					result, err := r.reverseDNSCache.Lookup(ip)
+					if err != nil {
+						r.Log.Errorf("lookup error: %v", err)
```

Comment:
https://github.com/influxdata/telegraf/wiki/Logging#style

Reply:
So, just capitalize it?